central/processlisteningonport/datastore/datastore_impl.go

```diff
@@ -40,6 +40,50 @@ func newDatastoreImpl(
 	}
 }
 
+func convertPlopFromStorageToPlopFromSensor(plopStorage *storage.ProcessListeningOnPortStorage) *storage.ProcessListeningOnPortFromSensor {
+	return &storage.ProcessListeningOnPortFromSensor{
+		Port:		plopStorage.Port,
+		Protocol:	plopStorage.Protocol,
+		Process:	plopStorage.Process,
+		CloseTimestamp:	plopStorage.CloseTimestamp,
+		// Storage does not have ClusterId which PlopFromSensor has, but this does not seem
+		// to be a problem. Might want to examine this more.
+		// ClusterId:	plopStorage.ClusterId,
+	}
+}
+
+func (ds *datastoreImpl) getUnmatchedPlopsFromDB(ctx context.Context) []*storage.ProcessListeningOnPortStorage {
+        plopsFromDB := []*storage.ProcessListeningOnPortStorage{}
+        ds.WalkAll(ctx,
+                func(plop *storage.ProcessListeningOnPortStorage) error {
+			if plop.ProcessIndicatorId == "" {
+				plopsFromDB = append(plopsFromDB, plop)
+			}
+			return nil
+                })
+
+        return plopsFromDB
+}
+
+func (ds *datastoreImpl) addUnmatchedProcesses(ctx context.Context, portProcesses []*storage.ProcessListeningOnPortFromSensor) []*storage.ProcessListeningOnPortFromSensor {
+
+	unmatchedIds := make([]string, 0)
+	unmatchedPLOPs := ds.getUnmatchedPlopsFromDB(ctx)
+
+	for _, val := range unmatchedPLOPs {
+		unmatchedIds = append(unmatchedIds, val.Id)
+		plop := convertPlopFromStorageToPlopFromSensor(val)
+		portProcesses = append(portProcesses, plop)
+	}
+
+	// Unmatched plop objects are deleted and added back in to avoid duplicate rows.
+	// Deleting unmatched plops is not the most efficient solution.
+	// Instead plopObjects should be set correctly in AddProcessListeningOnPort.
+	ds.storage.DeleteMany(ctx, unmatchedIds)
+
```

Comment:
Unmatched plops should not be allowed to persist in the table forever. There should probably be another column in the table that specifies when the row was first inserted. Unmatched plops could then be removed say after 90 seconds.